source/beamline/BioXAS/BioXASCarbonFilterFarmActuatorPositionControl.cpp

```diff
@@ -71,38 +71,26 @@ void BioXASCarbonFilterFarmActuatorPositionControl::setMotor(CLSMAXvMotor *newCo
 {
 	if (motor_ != newControl) {
 
-		if (motor_)
-			removeChildControl(motor_);
+            removeChildControl(motor_);
+            motor_ = newControl;
+            addChildControl(motor_);
+            updateStates();
 
-		motor_ = newControl;
-
-		if (motor_)
-			addChildControl(motor_);
-
-		updateStates();
-
-		emit motorChanged(motor_);
+            emit motorChanged(motor_);
 	}
 }
 
 void BioXASCarbonFilterFarmActuatorPositionControl::setStatus(AMControl *newControl)
 {
 	if (status_ != newControl) {
 
-		if (status_)
-			removeChildControl(status_); // disconnects all signals.
-
-		status_ = newControl;
-
-		if (status_) {
-			addChildControl(status_);
-
-			connect( status_, SIGNAL(valueChanged(double)), this, SLOT(onStatusControlValueChanged()) );
-		}
-
-		updateStates();
+            removeChildControl(status_); // disconnects all signals.
+            status_ = newControl;
+            addChildControl(status_);
+            connect( status_, SIGNAL(valueChanged(double)), this, SLOT(onStatusControlValueChanged()) );
```

Comment:
@mButlerDM as I mentioned in the other comments, this won't change the behavior. We only need to connect the signal/slot when the new status_ can be added as a child. If it can't, then either status_ is null or it is already added (and the signal/slot is connected already).